docker-official-images/main_test.go

```diff
@@ -0,0 +1,23 @@
+package main
+
+import (
+	"log"
+	"os"
+	"testing"
+)
+
+// TestMainFunction provides a basic test for the main function.
+func TestMainFunction(t *testing.T) {
```

Comment:
I think that one very important test to make here is to get the official images file from the official repo ([sonarqube](https://github.com/docker-library/official-images/blob/3501d02bf2a2551f950063778437206f347207f9/library/sonarqube)) and use it as a fixture.
Then, create the necessary test data (e.g. a test `active_versions.json`) that will generate that file, and compare what we generate with the fixture.

Currently, the official file has a slightly different structure (global GitCommit, duplicated Architectures for all the images, etc), but we can use that as a base for a fixture.

WDYT?